dates.py

```diff
@@ -1,10 +1,59 @@
 import re
+import logging
+
 from google.cloud import storage
+from google.auth.exceptions import DefaultCredentialsError
+
+# Create/configure stream handler
+sh = logging.StreamHandler()
+fmt = logging.Formatter('%(name)s.py - %(levelname)s - %(message)s')
+sh.setFormatter(fmt)
+
+# Register stream handler
+logger = logging.getLogger(__name__)
+logger.addHandler(sh)
 
 GCS_BUCKET = 'httparchive'
 
-def get_dates():
+# When running the site locally and unable to authenticate with GCS bypass
+# loading dates from GCS.
+LOAD_DATES_FROM_GCS = True
+try:
 	gcs = storage.Client()
+except DefaultCredentialsError:
+	logger.warning('Unable to authenticate to Google Cloud Storage.')
+	LOAD_DATES_FROM_GCS = False
+
+	def mock_get_dates(start_year, end_year, month_delta=0):
+		''' Generate mock dates to the format the site frontend expects them'''
+		current_year = start_year
+		months = []
+		for month in range(1, ((end_year - start_year) * 12) + month_delta):
+			current_month_in_year = month % 12
+			if current_month_in_year == 0:
+				current_month_in_year = 12
+
+			if current_month_in_year < 10:
+				current_month_in_year = "0{}".format(current_month_in_year)
+
+			months.append('{}_{}_{}'.format(current_year, current_month_in_year, '01'))
```

Comment:
Out of curiosity, is there an advantage to using `format` as opposed to string formatting, eg:

```py
months.append(`%s_%s_01` % (current_year, current_month_in_year))
```

Reply:
Just a personal preference as I find it more expressive in some cases - I could go either way with this :)